### PR TITLE
pd: support dpa3 with paddle backend

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
support dpa3 with paddle backend(eager mode)

### 1. training curve

pt_vs_pd

### 2. accuracy

<details>
  <summary>torch</summary>

  image

</details>

<details>
  <summary>paddle(slightly better than torch)</summary>

  image

</details>

### 3. The main modifications in this PR include:

1. Added DPA-3 code and related modules based on the Paddle backend.
2. Added the EnergyHessianStdLoss module based on the Paddle backend.
3. Discovered that Paddle’s ParameterList does not support assignment of Tensors using the equals sign. Therefore, I added support for this feature at <https://github.com/PaddlePaddle/Paddle/pull/72190>. However, considering version compatibility, deepmd still uses paddle.assign for assignments.
4. Fixed an issue in env_mat_stat.py where the return type was Tensor instead of float.
5. The SiLUT used APIs from the numpy series that do not support paddle.Tensor, so I replaced them with Paddle’s native APIs. Additionally, to temporarily bypass issues with dynamic-to-static control flow, I changed the if-else branch in SiLUT.forward to a single branch.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new descriptor, DPA3, for advanced molecular simulations, including its integration and public availability.
  - Added support for a new graph-based neural network layer and descriptor block for RepFlow calculations.
  - Enabled Hessian loss computation for enhanced training capabilities.
  - Added new learning rate utility.

- **Bug Fixes**
  - Improved tensor shape handling and assignments for better compatibility and stability.

- **Tests**
  - Added comprehensive tests for the new DPA3 descriptor, including consistency, JIT, and multitask scenarios.
  - Expanded test coverage for model permutation and smoothness with DPA3.
  - Enhanced tests for DPA2 with CINN compiler support.

- **Refactor**
  - Standardized tensor shape definitions and updated method signatures for improved clarity and type safety.

- **Chores**
  - Updated public interfaces to include new features and descriptors.

### CODE REVIEW

**Comment by HydrogenSulfate** [comment]
@njzjz This PR is ready for review

**Review by iProzd — COMMENTED** [review]
@HydrogenSulfate Note that this PR not only adds DPA3 descriptor, but also modifies other parts (such as **hessian loss**, **silut activation** and other improvements). Please at least keep a list of modifications in this PR comment. The main modifications are LGTM.

**Comment by HydrogenSulfate** [comment]
> @HydrogenSulfate Note that this PR not only adds DPA3 descriptor, but also modifies other parts (such as **hessian loss**, **silut activation** and other improvements). Please at least keep a list of modifications in this PR comment. The main modifications are LGTM.

Thanks for comments, I have added the summary of modification. 
image

**Comment by njzjz** [comment]
@coderabbitai resolve
